src/lib/cdev/posix/cdev_platform.cpp

```diff
@@ -34,68 +34,88 @@
 
 #include "cdev_platform.hpp"
 
-#include <string>
-#include <map>
-
-#include "vfile.h"
 #include "../CDev.hpp"
 
 #include <px4_platform_common/log.h>
 #include <px4_platform_common/posix.h>
 #include <px4_platform_common/time.h>
 
-using namespace std;
-
 const cdev::px4_file_operations_t cdev::CDev::fops = {};
 
 pthread_mutex_t devmutex = PTHREAD_MUTEX_INITIALIZER;
 pthread_mutex_t filemutex = PTHREAD_MUTEX_INITIALIZER;
 
-#define PX4_MAX_FD 350
-static map<string, void *> devmap;
-static cdev::file_t filemap[PX4_MAX_FD] = {};
+struct px4_dev_t {
+	char *name{nullptr};
+	cdev::CDev *cdev{nullptr};
 
-extern "C" {
+	px4_dev_t(const char *n, cdev::CDev *c) : cdev(c)
+	{
+		name = strdup(n);
+	}
 
-	static cdev::CDev *getDev(const char *path)
+	~px4_dev_t()
 	{
-		PX4_DEBUG("CDev::getDev");
+		free(name);
+	}
+private:
+	px4_dev_t() = default;
+};
 
-		pthread_mutex_lock(&devmutex);
+static px4_dev_t *devmap[128] {};
 
-		auto item = devmap.find(path);
+#define PX4_MAX_FD 128
+static cdev::file_t filemap[PX4_MAX_FD] {};
 
-		if (item != devmap.end()) {
-			pthread_mutex_unlock(&devmutex);
-			return (cdev::CDev *)item->second;
-		}
+class VFile : public cdev::CDev
+{
+public:
+	VFile(const char *fname, mode_t mode) : cdev::CDev(fname) {}
+	~VFile() override = default;
 
-		pthread_mutex_unlock(&devmutex);
+	ssize_t write(cdev::file_t *handlep, const char *buffer, size_t buflen) override
+	{
+		// ignore what was written, but let pollers know something was written
+		poll_notify(POLLIN);
+		return buflen;
+	}
+};
 
-		return nullptr;
+static cdev::CDev *getDev(const char *path)
+{
+	pthread_mutex_lock(&devmutex);
+
+	for (const auto &dev : devmap) {
+		if (dev && (strcmp(dev->name, path) == 0)) {
+			pthread_mutex_unlock(&devmutex);
+			return dev->cdev;
+		}
 	}
 
-	static cdev::CDev *get_vdev(int fd)
-	{
-		pthread_mutex_lock(&filemutex);
-		bool valid = (fd < PX4_MAX_FD && fd >= 0 && filemap[fd].vdev);
-		cdev::CDev *dev;
+	pthread_mutex_unlock(&devmutex);
 
-		if (valid) {
-			dev = (cdev::CDev *)(filemap[fd].vdev);
+	return nullptr;
+}
 
-		} else {
-			dev = nullptr;
-		}
+static cdev::CDev *getFile(int fd)
+{
+	pthread_mutex_lock(&filemutex);
+	cdev::CDev *dev = nullptr;
 
-		pthread_mutex_unlock(&filemutex);
-		return dev;
+	if (fd < PX4_MAX_FD && fd >= 0 && filemap[fd].cdev) {
```

Comment:
You can even remove that check:
```suggestion
	if (fd < PX4_MAX_FD && fd >= 0) {
```